### PR TITLE
[Share to IG] Fix crash share to IG C-1612

### PR DESCRIPTION
**Pull request by nicoback2**
### Description
Share to IG feature was crashing the app due to a memory issue: https://sentry.io/organizations/audius/issues/3491156239/?query=is%3Aunresolved&referrer=issue-stream
This was only happening on release builds which is why I didn't see it when developing locally.
Fixed by moving the dominant color calculation outside of a react-native-thread and into just a normal async function. Seemed to have no impact on the loading time or the UI interactions/music playback etc.

From react-native-thread repo:
```
The main tradeoff of using this library is memory usage, as creating new JS processes can have significant overhead. Be sure to benchmark your app's memory usage and other resources before using this library! Alternative solutions include using runAfterInteractions or the [Interaction Manager](https://facebook.github.io/react-native/docs/interactionmanager.html), and I recommend you investigate those thoroughly before using this library.
```
Maybe worth trying `runAfterInteractions` if we do end up seeing issues with this change.
I left the react-native-thread stuff in so we can get this feature into QA quicker and in case we want to figure out how to make it work.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by raymondjacobson** [comment]
Defer to @dylanjeffers 's review on the actual code, but LGTM -- i'm curious if we can figure out a way to reproduce this more easily locally?

https://stackoverflow.com/a/4717157

**Comment by nicoback2** [comment]
> Defer to @dylanjeffers 's review on the actual code, but LGTM -- i'm curious if we can figure out a way to reproduce this more easily locally?
> 
> https://stackoverflow.com/a/4717157

Oh interesting yeah this might be useful. I was able to repro it locally by just running the app with the release config `"react-native run-ios --device \"Poop\" --configuration Release"` (Poop is my phone lol)

**Review by sliptype — APPROVED** [review]

**Review by dylanjeffers — APPROVED** [review]
this all def makes sense to me! would be interested to check out that memory warning thing ray pointed out, but i agree that ideally we don't need threads.

**Comment by dylanjeffers** [comment]
@nicoback small nit, re pr titles, are we doing [C-blah] first or [Story] with C-blah at the end?

**Comment by nicoback2** [comment]
> @nicoback q re pr titles, are we doing [C-blah] first or [Story] with C-blah at the end?

oh idk im super inconsistent lol, wdyt?

**Pull request merged**
